test/tests/props.js

```diff
@@ -87,6 +87,53 @@ describe('zoid props cases', () => {
         });
     });
 
+    it('should merge props definition with input props before passing to decorator', () => {
+        return wrapPromise(({ expect }) => {
+            window.__component__ = () => {
+                return zoid.create({
+                    tag:    'test-prop-value-container-url-param-bug',
+                    url:    'mock://www.child.com/base/test/windows/child/index.htm',
+                    domain: 'mock://www.child.com',
+                    props:  {
+                        account: {
+                            type:       'string',
+                            queryParam: false,
+                            required:   true,
+                            value:      () => undefined
```

Comment:
This test case is strange to me. The intention is that you should either:

a) Pass `value` when defining the prop
b) Pass the value when instantiating the component

...never both at the same time.

Reply:
Just tested, if I change this to:

```javascript
required: false,
default: () => undefined
```

-- then the test passes as expected.

Reply:
This is the scenario that was unaccounted for in the bug

Reply:
> This is the scenario that was unaccounted for in the bug

Yeah, so it looks like messaging-components is passing:

```javascript
required: true,
value: () => undefined
```

This is not correct, you can't have a "required but undefined" prop.

### Correct

```javascript
required: true,
value: () => 'foobar'
```

```javascript
required: false,
default: () => 'foobar'
```

```javascript
required: false,
default: () => undefined
```

### Incorrect

```javascript
required: true,
value: () => undefined
```

```javascript
required: true,
default: () => 'foobar'
```

```javascript
required: true,
default: () => undefined
```

Reply:
So if the props in messaging-components are potentially not defined, the correct way to specify that is to pass:

```javascript
required: false,
default: () => whatever
```

`required: false` -> The value does not need to be passed

`default: () => whatever` -> If there is a viable default it will be used.

Reply:
Actually I think what might be happening now that I see more about how `inputProps` and the `value` function is that on:

```
prop: {
  foo: { 
    value: ({props}) => prop.foo // Is this props param meant to contain the `inputProps`?
  }
}
```

Reply:
>  Is this props param meant to contain the `inputProps`

Yeah, I think that's the real bug in zoid here. Good catch.

Reply:
Okay I think I have a better idea of what's going on and what I'm trying to say. This mix up is definitely some confusion I had around the value function and the input prop behaviors. It looks like in the value function of a propDef prior to the last zoid release would give a `props` key that contained the values from `inputProps`, but after the latest release it only contains `existingProps`.

For messaging this means that `account` from `inputProps` is not present in the following's props parameter [code](https://github.com/paypal/paypal-messaging-components/blob/f0cd8d1e9b6dce5ad4602ad559e3f4f4824f10d5/src/zoid/message/validation.js#L55).

So I think the "bug" is there and hopefully the updated test shows this a little better. Running locally I can confirm this change in behavior, however if zoid was not intended to behave this way then that might be a different story. 

Regardless I'm basically learning zoid as I dive through this so bear with me

Reply:
Actually I'm still a little confused here. I'm writing a few quick test cases to verify this behavior and it's working as expected for me. Hold fire, will share a branch